src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        MachineProducer<? extends Machine> producer;
+        if (type == Bulldozer.class) {
+            producer = new BulldozerProducer();
+        } else if (type == Truck.class) {
+            producer = new TruckProducer();
+        } else if (type == Excavator.class) {
+            producer = new ExcavatorProducer();
+        } else {
+            return new ArrayList<>();
+        }
+        return (List<T>) producer.get();
```

Comment:
The cast `(List<T>) producer.get()` may lead to a `ClassCastException` at runtime if the types do not match exactly. Ensure that the `MachineProducer` implementations return the correct type to avoid this issue.